requests/script.js

```diff
@@ -0,0 +1,471 @@
+const API_BASE_URL = window.API_BASE_URL;
+const requestContainer = document.getElementById(REQUEST_CONTAINER_ID);
+const params = new URLSearchParams(window.location.search);
+const isDev = DEV_FEATURE_FLAG;
+const loader = document.querySelector('.container__body__loader');
+const startLoading = () => loader.classList.remove('hidden');
+const stopLoading = () => loader.classList.add('hidden');
+let isDataLoading = false;
+let oooTabLink = document.getElementById(OOO_TAB_ID);
+let currentReqType = OOO_REQUEST_TYPE;
+let selected__tab__class = 'selected__tab';
+let statusValue = null;
+let sortByValue = null;
+
+oooTabLink.addEventListener('click', function () {
+  oooTabLink.classList.add(selected__tab__class);
+  currentReqType = OOO_REQUEST_TYPE;
+  changeFilter();
+  renderOooRequestCards({ state: statusValue, sort: sortByValue });
+});
+
+async function getOooRequests(query = {}) {
+  let finalUrl = API_BASE_URL + '/requests' + getOooQueryParamsString(query);
+  if (isDev) {
+    finalUrl =
+      API_BASE_URL + '/requests' + getOooQueryParamsString(query) + '&dev=true';
+  }
+
+  try {
+    const res = await fetch(finalUrl, {
+      credentials: 'include',
+    });
+
+    if (res.ok) {
+      const data = await res.json();
+      return data;
+    }
+
+    if (res.status === 401) {
+      showMessage('ERROR', ErrorMessages.UNAUTHENTICATED);
+      return;
+    }
+
+    if (res.status === 403) {
+      showMessage('ERROR', ErrorMessages.UNAUTHORIZED);
+      return;
+    }
+
+    if (res.status === 404) {
+      showMessage('ERROR', ErrorMessages.OOO_NOT_FOUND);
+      return;
+    }
+
+    showMessage('ERROR', ErrorMessages.SERVER_ERROR);
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+function showMessage(type, message) {
+  const p = document.createElement('p');
+  const classes = ['request__message'];
+  if (type === 'ERROR') {
+    classes.push('request__message--error');
+  }
+  p.classList.add(...classes);
+  p.textContent = message;
+  requestContainer.innerHTML = '';
+  requestContainer.appendChild(p);
+}
+
+const changeFilter = () => {
+  requestContainer.innerHTML = '';
+};
+
```

Comment:
`showMessage` function is relatively concise, using a switch-case statement could offer more clarity, especially if there are multiple conditions to handle. 

By using a switch-case statement, the code becomes more structured and easier to follow, especially when handling multiple message types. Also, it enhances maintainability, as adding new message types in the future will require minimal changes to the function.

Reply:
Changes done. Please check and thanks for such a detailed explaination